Omit base64 image fields from cards when sending to matrix (and hence AI bot)

habdelra opened the pull request:
This PR is based on PR https://github.com/cardstack/boxel/pull/1060. Please review and merge that one first.

This PR strips out base64 image fields when sending the cards to matrix, which in turn prevents the aibot from having to deal with this field. OpenAI doesn't support base64 encoding (I asked it, and it confirmed as such), so these fields take up unnecessary space in our prompts to open ai.

habdelra committed:
Omit base64image field from cards sent to matrix (and hence AI bot)

backspace reviewed (approved):
(I changed the base branch to review this so I could see only the relevant changes, will change back after submitting this)

Looks good to me, I wonder if we might want a more general way to mark certain fields as not for serialisation, but that can definitely stay a future consideration.

ETA: I think the Percy diffs would be fixed by merging from `main`.

jurgenwerk commented on packages/matrix/tests/messages.spec.ts:
```diff
@@ -168,25 +173,70 @@ test.describe('Room messages', () => {
     page,
   }) => {
-    const testCard = `${testHost}/mango-puppy`; // this is a 153KB card
+    const testCard = `${testHost}/big-card`; // this is a 153KB card
```
Is the comment still valid for this file?

habdelra replied:
I’ll update it

habdelra replied:
actually this comment is still correct. the big-card uses the same data as the previous mango-puppy card

jurgenwerk reviewed (approved):
👍

habdelra commented:
> (I changed the base branch to review this so I could see only the relevant changes, will change back after submitting this)
> 
> Looks good to me, I wonder if we might want a more general way to mark certain fields as not for serialisation, but that can definitely stay a future consideration.
> 
> ETA: I think the Percy diffs would be fixed by merging from `main`.

Yeah, that’s why I added an `omitFields` option to our serializeCard api function. It works based on field type

habdelra committed:
Merge remote-tracking branch 'origin/main' into cs-6599-avoid-sending-base64-data-to-openai-api-update-to-aibot

IanCal commented:
Note for history - we don't want to pass these straight through, and when it comes to actually using images with a multimodal LLM we'll want to pass them over as a custom field so filtering them here makes sense. Also right now we're using the openai models and there is a multimodal option but that model cannot currently do any function calling or restricted output so isn't usable with our assistant right now.
